src/sortStudents.ts

```diff
@@ -1,16 +1,68 @@
-
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyArray: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name: {
+      copyArray.sort((a, b) => a.name.localeCompare(b.name));
+      break;
+    }
+
+    case SortType.Surname: {
+      copyArray.sort((a, b) => a.surname.localeCompare(b.surname));
+      break;
+    }
```

Comment:
```suggestion
    case SortType.Name:
    case SortType.Surname:
      copyArray.sort((firstStudent: Student, secondStudent: Student) => (
        firstStudent[sortBy].localeCompare(secondStudent[sortBy])
      ));
      break;
```
checklist pt 3 - [CODE STYLE] - DRY. Combine logic for sorting by fields with same data types.